apiserver/xdsserverv3/server.go

```diff
@@ -445,17 +451,97 @@ func generateServiceDomains(serviceInfo *ServiceInfo) []string {
 	return resDomains
 }
 
-func makeVirtualHosts(services []*ServiceInfo) []types.Resource {
+func makeLocalRateLimit(conf []*model.RateLimit) map[string]*anypb.Any {
+	filters := make(map[string]*anypb.Any)
+	if conf != nil {
+		rateLimitConf := &lrl.LocalRateLimit{
+			StatPrefix: "http_local_rate_limiter",
+			// TokenBucket: &envoy_type_v3.TokenBucket{
+			// 	MaxTokens:    rule.Amounts[0].MaxAmount.Value,
+			// 	FillInterval: rule.Amounts[0].ValidDuration,
+			// },
+		}
+		rateLimitConf.FilterEnabled = &core.RuntimeFractionalPercent{
+			RuntimeKey: "local_rate_limit_enabled",
+			DefaultValue: &envoy_type_v3.FractionalPercent{
+				Numerator:   uint32(100),
+				Denominator: envoy_type_v3.FractionalPercent_HUNDRED,
+			},
+		}
+		rateLimitConf.FilterEnforced = &core.RuntimeFractionalPercent{
+			RuntimeKey: "local_rate_limit_enforced",
+			DefaultValue: &envoy_type_v3.FractionalPercent{
+				Numerator:   uint32(100),
+				Denominator: envoy_type_v3.FractionalPercent_HUNDRED,
+			},
+		}
+		for _, c := range conf {
+			rlRule := c.Rule
+			rlLabels := c.Labels
+			if rlRule == "" {
+				continue
+			}
+			rule := new(api.Rule)
+			if err := json.Unmarshal([]byte(rlRule), rule); err != nil {
+				log.Errorf("unmarshal local rate limit rule error,%v", err)
+				continue
+			}
+			if err := json.Unmarshal([]byte(rlLabels), &rule.Labels); err != nil {
+				log.Errorf("unmarshal local rate limit labels error,%v", err)
+			}
+
+			// 跳过全局限流配置
+			if rule.Type == api.Rule_GLOBAL || rule.Disable.Value {
+				continue
+			}
+
+			for _, amount := range rule.Amounts {
+				descriptor := &envoy_extensions_common_ratelimit_v3.LocalRateLimitDescriptor{
+					TokenBucket: &envoy_type_v3.TokenBucket{
+						MaxTokens:    amount.MaxAmount.Value,
+						FillInterval: amount.ValidDuration,
+					},
+				}
+				var entries []*envoy_extensions_common_ratelimit_v3.RateLimitDescriptor_Entry
```

Comment:
entries := make([]xxx, len(rule.Labels), len(rule.Labels))
pos := 0
for k, v := range rule.Labels {
  entries[pos] = xxx
  pos ++
}